### PR TITLE
r/iam_saml_provider - add tagging support + validations

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #17594

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSIAMSamlProvider_'
--- PASS: TestAccAWSIAMSamlProvider_disappears (26.74s)
--- PASS: TestAccAWSIAMSamlProvider_basic (58.14s)
--- PASS: TestAccAWSIAMSamlProvider_tags (84.33s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSIAMSamlProvider_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSIAMSamlProvider_ -timeout 120m
=== RUN   TestAccAWSIAMSamlProvider_basic
=== PAUSE TestAccAWSIAMSamlProvider_basic
=== RUN   TestAccAWSIAMSamlProvider_tags
=== PAUSE TestAccAWSIAMSamlProvider_tags
=== RUN   TestAccAWSIAMSamlProvider_disappears
=== PAUSE TestAccAWSIAMSamlProvider_disappears
=== CONT  TestAccAWSIAMSamlProvider_basic
=== CONT  TestAccAWSIAMSamlProvider_disappears
=== CONT  TestAccAWSIAMSamlProvider_tags
--- PASS: TestAccAWSIAMSamlProvider_disappears (12.28s)
--- PASS: TestAccAWSIAMSamlProvider_basic (24.49s)
--- PASS: TestAccAWSIAMSamlProvider_tags (32.42s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	32.505s
```

File: aws/resource_aws_iam_saml_provider_test.go [hunk 1]
```diff
@@ -95,6 +96,49 @@ func TestAccAWSIAMSamlProvider_basic(t *testing.T) {
...
 }
 
+func TestAccAWSIAMSamlProvider_tags(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_iam_saml_provider.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIAMSamlProviderDestroy,
```
Comment (bflad) [inline aws/resource_aws_iam_saml_provider_test.go:106]:
```suggestion
		PreCheck:     func() { testAccPreCheck(t) },
		ErrorCheck:   testAccErrorCheck(t, iam.EndpointsID),
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckIAMSamlProviderDestroy,
```

File: website/docs/r/iam_saml_provider.html.markdown [hunk 2]
```diff
@@ -32,6 +32,7 @@ In addition to all arguments above, the following attributes are exported:
 
 * `arn` - The ARN assigned by AWS for this provider.
 * `valid_until` - The expiration date and time for the SAML provider in RFC1123 format, e.g. `Mon, 02 Jan 2006 15:04:05 MST`.
+* `tags` - Key-value map of tags for the IAM SAML provider.
```
Comment (bflad) [inline website/docs/r/iam_saml_provider.html.markdown:35]: Argument, not just attribute 👍  Will fix on merge.

**Review by bflad — APPROVED** [review]
Thank you, @DrFaust92 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSIAMSamlProvider_disappears (9.84s)
--- PASS: TestAccAWSIAMSamlProvider_basic (20.71s)
--- PASS: TestAccAWSIAMSamlProvider_tags (29.18s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSIAMSamlProvider_disappears (16.38s)
--- PASS: TestAccAWSIAMSamlProvider_basic (34.37s)
--- PASS: TestAccAWSIAMSamlProvider_tags (45.99s)
```

**Pull request merged**